lib/src/entrypoint.dart

```diff
@@ -224,6 +224,21 @@ class Entrypoint {
       {List<String> useLatest,
       bool dryRun = false,
       bool precompile = false}) async {
+    // We require an SDK constraint lower-bound as of Dart 2.12.0
+    final dartSdkConstraint = root.pubspec.sdkConstraints['dart'];
+    if (dartSdkConstraint is! VersionRange ||
+        (dartSdkConstraint is VersionRange && dartSdkConstraint.min == null)) {
+      throw DataException('''
+pubspec.yaml has no lower-bound SDK constraint.
+You should edit pubspec.yaml to contain an SDK constraint:
+
+environment:
+  sdk: '>=${sdk.version} <${sdk.version.nextBreaking}'
```

Comment:
This will suggest `>=2.12.0`, right? That will cause failures for most projects that didn't have an SDK constraint to begin with. Should we consider using a fixed lower-bound?

Second, I wonder if this works for non-stable builds; doesn't it end up suggesting a stable version lower bound for a release that still might be in dev or beta only?

Reply:
This will suggest the current SDK version as lower-bound (I think this SDK version includes prerelease information, I should add a test for that).

```
pubspec.yaml has no lower-bound SDK constraint.
You should edit pubspec.yaml to contain an SDK constraint:

environment:
  sdk: '>=${sdk.version} <${sdk.version.nextBreaking}'

See https://dart.dev/go/sdk-constraint
Use if your code is not migrated to null-safety yet.

environment:
  sdk: '>=2.10.0 <${sdk.version.nextBreaking}'

See <link-to-null-safety-migration-guide>
```

It just seemed like that was a rather lengthy message. I suspect `https://dart.dev/go/sdk-constraint` will explain opting-in/out of null-safety.

Reply:
I suggest we fix the lower bound at `2.10`

Reply:
```
pubspec.yaml has no lower-bound SDK constraint.
You should edit pubspec.yaml to contain an SDK constraint:

environment:
  sdk: '>=2.10 <${sdk.version.nextBreaking}'

For more details, see https://dart.dev/go/sdk-constraint
```

Reply:
:( isn't it a shame not promoting the latest and greatest?